src/core/ApolloClient.ts

```diff
@@ -228,6 +252,15 @@ export class ApolloClient<TCacheShape> implements DataProxy {
     this.queryDeduplication = queryDeduplication;
     this.defaultOptions = defaultOptions || Object.create(null);
     this.typeDefs = typeDefs;
+    this.devtoolsConfig = {
+      ...devtools,
+      enabled:
+        devtools?.enabled ??
+        connectToDevTools ??
+        (typeof window === "object" &&
+          (window as any).__APOLLO_CLIENT__ &&
+          __DEV__),
+    };
```

Comment:
```suggestion
    this.devtoolsConfig = {
      ...devtools,
      enabled:
        devtools?.enabled ||
        connectToDevTools,
    };
    if (this.devtoolsConfig.enabled === undefined) {
      this.devtoolsConfig.enabled = (typeof window === "object" &&
          (window as any).__APOLLO_CLIENT__ &&
          __DEV__)
    }
```

Technically, this would change `null` from meaning `false` to `default` - if we are not fine with that, we could do something like this instead.

Reply:
I think technically it was "unsupported" before (it allowed `undefined`, but not `null` as far as the types go), even though `null` was treated the same as `false`. But I'm also not opposed to being fully backwards compatible here so I think this change makes sense.